app/api/models/domain/TaxYear.scala

```diff
@@ -0,0 +1,99 @@
+/*
+ * Copyright 2023 HM Revenue & Customs
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package api.models.domain
+
+import play.api.libs.json.Writes
+
+import java.time.LocalDate
+
+/** Opaque representation of a tax year.
+  *
+  * @param value
+  *   A single-year representation, e.g. "2024" represents the tax year 2023-24.
+  */
+final case class TaxYear private (private val value: String) {
```

Comment:
I don't think TaxYear is ever used. Up to you but maybe could remove it along with the validations that apply to it?